Assets/_Functionalities/Wms/Scripts/WMSTileDataLayer.cs

```diff
@@ -0,0 +1,119 @@
+using Netherlands3D.CartesianTiles;
+using Netherlands3D.Rendering;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using UnityEngine.Rendering.Universal;
+
+namespace Netherlands3D.Twin
+{
+    public class WMSTileDataLayer : ImageProjectionLayer
+    {      
+
+        public int RenderIndex 
+        { 
+            get => renderIndex;
+            set
+            {
+                int oldIndex = renderIndex;
+                renderIndex = value;
+                if (oldIndex != renderIndex) 
+                    UpdateDrawOrderForChildren();
+            }
+        }
+
+        private int renderIndex = -1;
+
+        private string wmsUrl = "";
+        public string WmsUrl
+        {
+            get => wmsUrl;
+            set
+            {
+                wmsUrl = value;
+                if (!wmsUrl.Contains("{0}"))
+                    Debug.LogError("WMS URL does not contain a '{0}' placeholder for the bounding box.", gameObject);
+            }
+        }       
+
+        private void Awake()
+        {
+            //Make sure Datasets at least has one item
+            if (Datasets.Count == 0)
+            {
+                var baseDataset = new DataSet()
+                {
+                    maximumDistance = 3000,
+                    maximumDistanceSquared = 1000 * 1000
+                };
+                Datasets.Add(baseDataset);
+            }
+        }
+
+        protected override IEnumerator DownloadDataAndGenerateTexture(TileChange tileChange, Action<TileChange> callback = null)
+        {
+            var tileKey = new Vector2Int(tileChange.X, tileChange.Y);
+
+            if (!tiles.ContainsKey(tileKey))
+            {
+                onLogMessage.Invoke(LogType.Warning, "Tile key does not exist");
+                yield break;
+            }
+
+            Tile tile = tiles[tileKey];
+            var bboxValue = $"{tileChange.X},{tileChange.Y},{(tileChange.X + tileSize)},{(tileChange.Y + tileSize)}";
+            string url = wmsUrl.Replace("{0}", bboxValue);
+
+            UnityWebRequest webRequest = UnityWebRequestTexture.GetTexture(url);
+            tile.runningWebRequest = webRequest;
+            yield return webRequest.SendWebRequest();
+            tile.runningWebRequest = null;
+            if (webRequest.result != UnityWebRequest.Result.Success)
+            {
+                Debug.LogWarning($"Could not download {url}");
+                RemoveGameObjectFromTile(tileKey);
+                callback(tileChange);
```

Comment:
Als je hier een yield break zou hebben neergezet zou de else niet nodig zijn geweest en vermindert dat de code complexiteit (zie het concept van Return Early -https://szymonkrajewski.pl/why-should-you-return-early/- o.a. uit Object Calisthenics of Cyclomatic Complexity als het gaat om code kwaliteitsmetrieken)

Reply:
Ik denk dat er standaard gebruik moet gemaakt worden van slimme / early returns, dus de reden voor deze informatie begrijp ik niet. Verder heb ik geprobeerd de stijl van de WFSGeoJSONTileDataLayer aan te houden zodat we consistentie hebben (en eigenlijk biijna overal in de codebase waar we een unitywebrequest aanroepen word). 
Ik zie wat ik vergeten ben dat is de callback op het eind te zetten zoals op andere plekken. Heb deze meegenomen, wel goed gezien!